cli/doctest_runner.rs

```diff
@@ -0,0 +1,223 @@
+use regex::Regex;
+use std::path::{Path, PathBuf};
+
+use crate::fs as deno_fs;
+use crate::installer::is_remote_url;
+use crate::test_runner::is_supported;
+
+pub struct DocTest {
+  // This removes repetition of imports in a file
+  imports: std::collections::HashSet<String>,
+  // This contains codes in an @example section with their imports removed
+  bodies: Vec<DocTestBody>,
+}
+
+struct DocTestBody {
+  caption: String,
+  line_number: usize,
+  path: PathBuf,
+  value: String,
+  ignore: bool,
+  is_async: bool,
+}
+
+pub fn prepare_doctest(
+  include: Vec<String>,
+  // root_path: &PathBuf,
+) -> Vec<DocTest> {
+  let include_paths: Vec<_> =
+    include.into_iter().filter(|n| !is_remote_url(n)).collect();
+
+  let mut prepared = vec![];
+
+  for path in include_paths {
+    let p = deno_fs::normalize_path(&Path::new(&path));
+    if p.is_dir() {
+      let test_files = crate::fs::files_in_subtree(p, |p| {
```

Comment:
style: you imported this path as `deno_fs`, should probably use that path consistently throughout the file